software/due-date-changes.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import re
+import sys
+import time
+
+from datetime import datetime
+from jira import JIRA
+
+
+def dump(obj):
+    for attr in dir(obj):
+        print("obj.%s = %r" % (attr, getattr(obj, attr)))
+
+
+def print_help(stream=sys.stderr):
+    help_msg = """Usage: {0} [--projects] LIST,OF,PROJECTS
+"""
+    stream.write(help_msg.format(sys.argv[0]))
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-projects", 
```

Comment:
I think this should be

```suggestion
    parser.add_argument("--projects", 
```

to match the help.

(Apparently argparse does actually let you do condor-style, single-dash long-options - though those kind of go against the unix/gnu norm...)

Reply:
I used the single dash option here because the other scripts use `-o <output-filename>` and I wanted to keep it all consistent. So I propose keeping this as is.

Reply:
I agree with @edquist on this one. You can provide a double-dashed option in addition to the single-dashed option, which is the convention:

```suggestion
    parser.add_argument("-p",
                        "--projects",
```

Reply:
> I used the single dash option here because the other scripts use `-o <output-filename>` and I wanted to keep it all consistent.

@markcoatsworth - first of all the help message says `--projects` [here](https://github.com/path-cc/metric-tools/pull/30/files#diff-53919459b3fb11c548b97fd68dfa7878d0a8a7c712413d486a1b879587b2ffd7R19).

Second, i am sorry to be the one to tell you, but condor has tainted your worldview, with it's single-dash long-options.

In the GNU coreutils world anyway (the core standard utilities that come with a linux distro), there are "short options" and "long options".  Short options consist of a single dash and a single letter, like `-o`.  Long options start with a double-dash and can be words or whatever, like `--projects`.

In general the fancy thing about short options is you can munge them if they don't take arguments.  So for instance `ls -lrt` is the same as `ls -l -r -t`.  Or for an osg throwback, `osg-test -vandamm` is the same as `osg-test -v -a -n -d -a -m -m` (which looks to have a redundant second `-a` and `-m`, but those add mnemonic value).

Anyway, my 2c is that the standard convention (outside of condor) is `-projects` would be interpreted as a bunch of short options munged together.  On the other hand, this is an internal script and so far you are its original and (up to this point) sole author, so i think it's fair to say you get special creative liberty with it.

But if you are just trying to make the `[-]-projects` option consistent with `-o`, i feel like it's my duty to inform you that the normal thing is 1 dash for single-letter options and two otherwise.

Reply:
Alright point taken, I didn't know we have the option to include both so I'm committing the suggested from @brianhlin